src/data_source/storage/sql.rs

```diff
@@ -600,22 +631,23 @@ impl PruneStorage for SqlStorage {
             .await?;
 
         if let Some(target_height) = target_height {
-            while height < target_height {
+            if height < target_height {
```

Comment:
I feel like we should save `target_height` in `self.pruner` when a run is in progress, so that all batches within a single "run" are moving towards the same target. If we reload target height each time, there is some risk that we never finish a run, because we return after the first batch, and by the time we are called again, the target height has moved forward just a little. Then we end up pruning really inefficiently one block at a time or so